glue_jupyter/bqplot/profile/layer_artist.py

```diff
@@ -0,0 +1,162 @@
+from __future__ import absolute_import, division, print_function
+
+import sys
+import warnings
+
+import numpy as np
+
+from glue.core import BaseData
+from glue.utils import defer_draw, nanmin, nanmax
+from glue.viewers.profile.state import ProfileLayerState
+from glue.core.exceptions import IncompatibleAttribute, IncompatibleDataException
+
+import bqplot
+import ipywidgets.widgets.trait_types as tt
+
+from glue_jupyter.compat import LayerArtist
+
+from ...link import link
+
+
+# FIXME: monkey patch ipywidget to accept anything
+tt.Color.validate = lambda self, obj, value: value
+
+
+class BqplotProfileLayerArtist(LayerArtist):
+
+    _layer_state_cls = ProfileLayerState
+
+    def __init__(self, view, viewer_state, layer_state=None, layer=None):
+
+        super().__init__(viewer_state, layer_state=layer_state, layer=layer)
+
+        # Watch for changes in the viewer state which would require the
+        # layers to be redrawn
+        self._viewer_state.add_global_callback(self._update_profile)
+        self.state.add_global_callback(self._update_profile)
+
+        self.view = view
+
+        self.line_mark = bqplot.Lines(scales=self.view.scales, x=[0, 1], y=[0, 1])
+
+        self.view.figure.marks = list(self.view.figure.marks) + [self.line_mark]
+
+        link((self.state, 'color'), (self.line_mark, 'colors'), lambda x: [x], lambda x: x[0])
+        link((self.state, 'alpha'), (self.line_mark, 'opacities'), lambda x: [x], lambda x: x[0])
+
+        self.line_mark.colors = [self.state.color]
+        self.line_mark.opacities = [self.state.alpha]
+
+    def _calculate_profile(self, reset=False):
+        try:
+            self._calculate_profile_thread(reset=reset)
+        except Exception:
+            self._calculate_profile_error(sys.exc_info())
+        else:
+            self._calculate_profile_postthread()
```

Comment:
The mention of threads is because this is adapted from the code in glue-core, which is multi-threaded. I'm trying to keep the code as similar as possible to the core code so that we can later try and reduce duplication.